src/client/SharedQueue.h

```diff
@@ -66,12 +66,45 @@ class DataHandler {
     void operator()() {
         while (true) {
             int enqueue_result = sharedQueue.enqueue();
-
             if (enqueue_result == 1) return;
         }
     }
 };
 
+class WarmUpRunner {
+  private:
+    SharedQueue &sharedQueue;
+    inline static std::mutex mutex;
+
+  public: 
+    inline static int warmupOperations; 
+
+    WarmUpRunner(SharedQueue& sharedQueue): sharedQueue(sharedQueue) {}
+
+    void operator()() {
+        while (true) {
+            std::unique_lock<std::mutex> lock(mutex);
+
+            if (warmupOperations == 0) return;
+            --warmupOperations;
+
+            lock.unlock();
+
+            Operation data = sharedQueue.dequeue();
+            if (data.key == "EOF") return;
```

Comment:
Would line 94 be resolved if we didn't have the extra newline at the end of seed files?

Reply:
Yes, we require some indiction from the dequeue function to the runners that we've reached EOF. I will add a derived class to Operation that specifically indicates this rather than using `data.key == "EOF"`